### PR TITLE
*.md: add workflow topic

### PR DESCRIPTION
**Pull request by vbatts**
This exists in a number of our repos already

Signed-off-by: Vincent Batts <vbatts@kinvolk.io>

### CODE REVIEW

File: topics/workflow.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Contribution Flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Create a topic branch from where you want to base your work (usually master).
+- Make commits of logical units.
```
Comment (joaquimrocha) [inline topics/workflow.md:6]:
```suggestion
- Make commits atomic (of logical and independent units).
- Ensure each commit builds successfully and independently from others (this is
  important when, e.g. reverting one commit or doing a git bisect).
```

**Review by joaquimrocha — COMMENTED** [review]
Added a suggestion.
Also, I don't think we have used `*: …` for commit messages and honestly I don't see much value in that. Normally if commits apply to many files I just move the context to the message rather than a prefix, something like:

```
Add workflow topic

This new topic brings the common workflow for contributing to our projects.
This commit also adds the new topic to the README's Development Topics' index.
```

**Comment by vbatts** [comment]
> Added a suggestion.
> Also, I don't think we have used `*: …` for commit messages and honestly I don't see much value in that. Normally if commits apply to many files I just move the context to the message rather than a prefix, something like:

It's fine if we haven't been using it, but it is a standard and expectation beyond even Kinvolk. And something I try to do personally. Whether it's a file name, package or say all source (i.e. `*.go: fix import paths`).

**Comment by vbatts** [comment]
updated PTAL

**Pull request merged**